src/pylorax/sysutils.py

```diff
@@ -130,3 +130,16 @@ def flatconfig(filename):
     config = UnquotingConfigParser()
     config.read_string(conftext)
     return config['main']
+
+def read_tail(path, size):
+    """Read up to `size` kibibytes from the end of a file"""
+    with open(path, "r") as f:
+        f.seek(0, 2)
+        end = f.tell()
+        if end < 1024 * size:
+            f.seek(0, 0)
+        else:
+            f.seek(end - (1024 * size))
```

Comment:
Calling `f.seek()` with anything except the return value from `f.tell()` is undefined for text files (https://docs.python.org/3/tutorial/inputoutput.html). This will probably work for your case, because you're seeking to the next line below (which should be mentioned in the docstring). It seems smelly to me, though.

To be honest, I'd scrap this whole function and just load the full file into memory as a string above. It's 2019, we can load a MB of text easily.

Reply:
Sounds fine to me. We can keep the existing get-last-n-bytes behavior by just assuming 1 byte per character.

Reply:
> Calling `f.seek()` with anything except the return value from `f.tell()` is undefined for text files (https://docs.python.org/3/tutorial/inputoutput.html). This will probably work for your case, because you're seeking to the next line below (which should be mentioned in the docstring). It seems smelly to me, though.

Oh geez. Apparently python3 changed that and no longer matches normal fseek behavior. Maybe we should open it as binary then?

> To be honest, I'd scrap this whole function and just load the full file into memory as a string above. It's 2019, we can load a MB of text easily.

The problem with that is python's garbage collector is sometime terrible about getting rid of things. We may be re-reading this same file repeatedly, so I'm worried about excess memory usage. We really ought to be able to properly tail a file in python in 2019 :)

Reply:
> The problem with that is python's garbage collector is sometime terrible about getting rid of things. We may be re-reading this same file repeatedly, so I'm worried about excess memory usage. We really ought to be able to properly tail a file in python in 2019 :)

Oh my. Good points. Let's open it in binary mode then. We just need to make sure that we're on the boundary of a code point, which seeking to the next `\n` will do.

Thanks.